tiledb/sm/query/query.cc

```diff
@@ -56,35 +59,51 @@ namespace sm {
 
 Query::Query(StorageManager* storage_manager, Array* array, URI fragment_uri)
     : array_(array)
+    , layout_(Layout::ROW_MAJOR)
     , storage_manager_(storage_manager)
-    , stats_(storage_manager_->stats()->create_child("Query"))
-    , reader_(stats_)
-    , writer_(stats_) {
-  assert(array != nullptr && array->is_open());
+    , has_coords_buffer_(false)
+    , has_zipped_coords_buffer_(false)
+    , coord_buffer_is_set_(false)
+    , coord_data_buffer_is_set_(false)
+    , coord_offsets_buffer_is_set_(false)
+    , data_buffer_name_("")
+    , offsets_buffer_name_("")
+    , disable_check_global_order_(false)
+    , sparse_mode_(false)
+    , fragment_uri_(fragment_uri) {
+  if (array != nullptr) {
+    assert(array->is_open());
+    array_schema_ = array->array_schema();
+
+    auto st = array->get_query_type(&type_);
+    assert(st.ok());
+
+    auto temp_stats = storage_manager_->stats()->create_child("Query");
+
+    if (type_ == QueryType::WRITE) {
+      stats_ = temp_stats->create_child("Writer");
```

Comment:
I believe we should still keep the `stats` object at the query class and not have the `Query` container the `reader`/`writer` stats. The stats have a hierarchy that is designed to match the class structure. Even if the `Query` stats are empty we should maintain this hierarchy if we are keeping the stats object.